src/js/components/award/shared/overview/ProgressBar.jsx

```diff
@@ -0,0 +1,560 @@
+            const circleStart = position - halfVisualizationHeight;
+            const circleEnd = position + halfVisualizationHeight;
+            return {
+                key: `${uniqueId(milestone.data)}`,
+                description: milestone.description,
+                className: milestone.className,
+                cx: milestone.cx || position,
+                cy: halfVisualizationHeight,
+                r: halfVisualizationHeight,
+                visualizationStart: milestone.visualizationStart || circleStart,
+                visualizationEnd: milestone.visualizationEnd || circleEnd,
+                display: false
+            };
+        });
+        this.setState({ milestoneData }, this.showMilestones);
+    }
+    // show milestones
+    showMilestones = () => {
+        const { milestoneData } = this.state;
+        const lastMilestoneIndex = milestoneData.length - 1;
+        const displayMilestoneData = milestoneData.map((milestone, index, array) => {
+            const milestoneObject = { ...milestone };
+            // ignore start and end milestones
+            if (index !== 0 && index !== lastMilestoneIndex) {
+                const startMilestoneEndPosition = array[0].visualizationEnd;
+                const endMilestoneStartPosition = array[lastMilestoneIndex].visualizationStart;
+                // verify current milestone start is after first milestone end and
+                // current milestone end is before last milestone start
+                if (
+                    (milestone.visualizationStart > startMilestoneEndPosition) &&
+                    (milestone.visualizationEnd < endMilestoneStartPosition)
+                ) {
+                    milestoneObject.display = true;
+                }
+                return milestoneObject;
+            }
+            milestoneObject.display = true;
+            return milestoneObject;
+        });
+        return this.setState({
+            milestoneData: displayMilestoneData
+        }, this.progressVerticalLineData);
+    }
+    // progress circle data
+    progressVerticalLineData = () => {
+        const { milestoneData, xScaleWithinCircles } = this.state;
+        const {
+            currentProgress,
+            domain,
+            descriptions,
+            heightOfProgressBar
+        } = this.props;
+        const position = xScaleWithinCircles(currentProgress);
+        const progressVerticalLineData = {
+            className: 'progress-bar-shapes__progress-vertical-line',
+            description: descriptions.progressVerticalLineDescription,
+            key: `progress-line`,
+            x1: position,
+            x2: position,
+            y1: 0,
+            y2: heightOfProgressBar,
+            visualizationStart: position,
+            visualizationEnd: position,
+            display: true
+        };
+        // show progress vertical line
+        milestoneData.forEach((milestone, index, array) => {
```

Comment:
If mutating an array, don't use `.forEach`. Instead, create a new array and use `.map`

Reply:
we are not mutating an array here. We are just updating the `progressVerticalLineDataObject` `display` property. We need to loop through the milestones.